pep-0570.rst

```diff
@@ -385,6 +385,202 @@ as a rule is not LL(1). This is the reason the rule needs to be included in
+
+Now users can call your function with *all* possible keyword parameters ::
+
+    >>> format_template(slow_template, name="Hilbert", template="fractal")
+    "After some slow computations, *Hilbert* has finished the *fractal* template"
+
+Some of the properties of positional-only parameters can be emulated using
+``*args`` and some manual processing. This has the obvious downside that you
+need to do the processing on each function and that the signature of the
+functions that you create will not reflect how it must be called. For
+instance, for the last example, the signature will be ::
+
+    >>> help(format_template)
+    Help on function format_template in module best_functions:
+
+    format_template(*args, **kwargs)
+
+but with positional-only parameters will be ::
+
+    >>> help(format_template)
+    Help on function format_template in module best_functions:
+
+    format_template(template, /, **kwargs)
+
+The second version is more explicit and clear and can benefit from systems
+that can generate the documentation automatically by reading the function
+signatures. Additionally, if you plan to use type annotations for your
+functions, using a single parameter instead of ``*args`` can be easier to
+deal with.
+
+Some other scenarios where positional-only parameters can appear are when
+you have a function that is implemented using the C-API. Functions that are
+implemented in C and exposed to Python can implement very easily
+positional-only parameters. Is common to provide as well a pure-Python
+fallback for these types of functions for users that cannot use the compiled
+version. Positional-only parameters offer a way of declaring in both
+interfaces how the function will behave when called and you can mirror
+easily the behaviour of the compiled function very easily from the Python
+side.
+
+Finally, positional-only parameters can give the advantage that all calls
+from users to your function will look the same, as the arguments needs to be
+provided always in the same order and without keywords. This can be useful
+if you have a function where there is a logical order to the arguments like
+the builtin ``range``.
+
+You can combine positional-only parameters with defaults and keyword-only
+parameters, for example::
+
+    class Function:
+        def integrate(self, start, stop=None, /, *, method="automatic"):
+            ....
+
+When calling this method, all of these forms are correct ::
+
+    function.integrate(0)
+    function.integrate(0, method="euler")
+    function.integrate(0, 100)
+    function.integrate(0, 100, method="euler")
+
+As you can see, positional-only arguments are specially convenient when you
+want to make sure that some of the interfaces you provide can be easily
+modified in the future without making backwards incompatible changes in your
+library. It allows you to have clear and explicit function signatures that
+express how arguments should be provided without the need of implementing
```

Comment:
Expresses? Some native speaker might help rewording this.

Reply:
Pablo's actually correct.